index.d.ts

```diff
@@ -10,12 +10,14 @@ declare const plugin: {
 		dom: Linter.LegacyConfig;
 		marko: Linter.LegacyConfig;
 		react: Linter.LegacyConfig;
+		svelte: Linter.LegacyConfig;
 		vue: Linter.LegacyConfig;
-		'flat/angular': Linter.FlatConfig;
-		'flat/dom': Linter.FlatConfig;
-		'flat/marko': Linter.FlatConfig;
-		'flat/react': Linter.FlatConfig;
-		'flat/vue': Linter.FlatConfig;
+		'flat/angular': Linter.Config<Linter.RulesRecord>;
+		'flat/dom': Linter.Config<Linter.RulesRecord>;
+		'flat/marko': Linter.Config<Linter.RulesRecord>;
+		'flat/react': Linter.Config<Linter.RulesRecord>;
+		'flat/svelte': Linter.Config<Linter.RulesRecord>;
+		'flat/vue': Linter.Config<Linter.RulesRecord>;
```

Comment:
note that these types should not have been changed - while `Linter.FlatConfig` is marked as deprecated in v9, it's the correct type to be using for v8 since `Linter.Config` there resolves to the legacy configuration.

For plugins like this which support both v8 and v9, we should be using the explicit `Linter.LegacyConfig` and `Linter.FlatConfig` types to ensure users of both receive the right types.

I have opened #976 to revert this change

Reply:
Indeed I didn't realise these were necessary for v8 compatibility

Thanks for the fix !